src/patterns/ec2-app.ts

```diff
@@ -0,0 +1,105 @@
+import { HealthCheck } from "@aws-cdk/aws-autoscaling";
+import { Certificate } from "@aws-cdk/aws-certificatemanager";
+import { ApplicationProtocol, ListenerAction } from "@aws-cdk/aws-elasticloadbalancingv2";
+import { Duration } from "@aws-cdk/core";
+import { GuAutoScalingGroup, GuUserData } from "../constructs/autoscaling";
+import type { GuStack } from "../constructs/core";
+import { GuArnParameter } from "../constructs/core";
+import type { AppIdentity } from "../constructs/core/identity";
+import { GuVpc, SubnetType } from "../constructs/ec2";
+import { GuInstanceRole } from "../constructs/iam";
+import {
+  GuApplicationListener,
+  GuApplicationLoadBalancer,
+  GuApplicationTargetGroup,
+} from "../constructs/loadbalancing";
+
+interface GuEc2AppProps extends AppIdentity {
+  userData: GuUserData | string;
+  publicFacing: boolean; // could also name it `internetFacing` to match GuApplicationLoadBalancer
+  applicationPort: number;
+}
+
+interface GuMaybePortProps extends Omit<GuEc2AppProps, "applicationPort"> {
+  applicationPort?: number;
+}
+
+export enum GuApplicationPorts {
+  Node = 3000,
+  Play = 9000,
+}
+
+/*
+This pattern is under development. Please don't attempt to use it yet.
+ */
+export class GuEc2App {
+  constructor(scope: GuStack, props: GuEc2AppProps) {
+    const id = (id: string) => `${props.app}${id}`;
+
+    scope.tags.setTag(id("App"), props.app);
+
+    const { app } = props;
+    const vpc = GuVpc.fromIdParameter(scope, id("VPC"), { app });
+    const privateSubnets = GuVpc.subnetsfromParameter(scope, { type: SubnetType.PRIVATE, app });
+
+    const certificateArn = new GuArnParameter(scope, id("CertArn"), {
+      description: "ARN of a TLS certificate to install on the load balancer",
+    });
+
+    const certificate = Certificate.fromCertificateArn(scope, id("Certificate"), certificateArn.valueAsString);
+
+    const asg = new GuAutoScalingGroup(scope, id("AutoScalingGroup"), {
+      vpc,
+      app: props.app,
```

Comment:
minor: we've destructured `app` above, so this can become:

```suggestion
      app,
```

Reply:
Good catch, done.